features/step_definitions/user_steps.rb

```diff
@@ -9,7 +11,10 @@
 
 Given /^(?:|I )am on (.+)$/ do |page_name|
   case page_name
-  when /^the profile page$/ then visit "/profile/1"
+  when /^the profile page$/ 
+    # We must use middleware to inject a proper session user_id, or else the page will fail authentication.
+    page.set_rack_session(:user_id => 1)
+    visit "/profile/1"
   when /^the edit page$/ then visit "profile/1/edit"
```

Comment:
Would we have to do the whole `set_rack_session` for edit page too, or no?

Reply:
I'm pretty sure all the profile page tests start with the show page, so it's fine.